lib/Dialect/Arc/ArcTypes.cpp

```diff
@@ -21,18 +21,61 @@ using namespace mlir;
 #define GET_TYPEDEF_CLASSES
 #include "circt/Dialect/Arc/ArcTypes.cpp.inc"
 
-unsigned StateType::getBitWidth() {
-  if (llvm::isa<seq::ClockType>(getType()))
+static std::optional<uint64_t> computeBitWidth(Type type) {
```

Comment:
maybe call it `computeLLVMBitWidth` or something to make it super clear that this includes alignment/padding

Reply:
Good idea!